### PR TITLE
More info for "unknown profile version" error message

### PR DESCRIPTION
**Pull request by mic704b**
Addresses enhancement request #2280 

# Changes
Adds version information to profile error message
Add profile error message to log

Error message:
version-error

About box:
about

Log messages
> 2020-01-19 03:31:18: "Database was open successfully"
2020-01-19 03:31:18: "Checking for database schema update..."
2020-01-19 03:31:18: "Current database version", "28"
2020-01-19 03:31:18: "Cannot convert database from version 28 to 27"

# Tests
Tested Electron app
Tested CliClient
Did not test on Mobile

# Comments
I did not implement @tessus suggestion in the issue concerning clipboard.  I investigated making the dialog box text "copy and paste" able, but not trivial. Automatic copy to clipboard is possible but may be unexpected for the user. Copy to clipboard via copy button would require creation of a new message type in the bridge. Please advise.

### CODE REVIEW

**Comment by tessus** [comment]
@mic704b thanks, my idea was just a suggestion to make it easier for people not to post screenshots of the About box. ;-) It's not essential. I certainly shouldn't stand in the way of merging this PR.

File: ReactNativeClient/lib/joplin-database.js [hunk 1]
```diff
@@ -313,7 +313,11 @@ class JoplinDatabase extends Database {
 
 		// currentVersionIndex < 0 if for the case where an old version of Joplin used with a newer
 		// version of the database, so that migration is not run in this case.
-		if (currentVersionIndex < 0) throw new Error('Unknown profile version. Most likely this is an old version of Joplin, while the profile was created by a newer version. Please upgrade Joplin at https://joplinapp.org and try again.');
+		if (currentVersionIndex < 0) {
+			this.logger().error(`Cannot convert database from version ${fromVersion} to ${existingDatabaseVersions[existingDatabaseVersions.length-1]}`);
+			const p = require('../package.json');
+			throw new Error(`Unknown profile version. Most likely this is an old version of Joplin, while the profile was created by a newer version. Please upgrade Joplin at https://joplinapp.org and try again.\n${p.name} version: ${p.version}\nProfile version: ${fromVersion}\nExpected version: ${existingDatabaseVersions[existingDatabaseVersions.length-1]}`);
```
Comment (tessus) [inline ReactNativeClient/lib/joplin-database.js]: Please use packageInfo here instead. As it is used in `app.js`.

**Comment by tessus** [comment]
@laurent22 did you want this on mobile and terminal as well? If so, please let me know and I'll update the text in #2280

File: ReactNativeClient/lib/joplin-database.js (hunk 1 above)
Comment (mic704b) [reply ReactNativeClient/lib/joplin-database.js]: That works for ElectronClient build, but not for CliClient.  It looks like packageInfo.js only exists in the ElectronClient build.  Is that right?

**Comment by mic704b** [comment]
> @mic704b thanks, my idea was just a suggestion 

Yes, but a good suggestion!

File: ReactNativeClient/lib/joplin-database.js (hunk 1 above)
Comment (tessus) [reply ReactNativeClient/lib/joplin-database.js]: Hmm, you might be right. I thought it was made available to all platforms, but I mixed it up with something else.

**Comment by mic704b** [comment]
@tessus  re the `desktop` tag, I think the error messages affect all platforms, not just `desktop`.

**Comment by tessus** [comment]
@mic704b yes, the error message affects all platforms. But the about box change is only on desktop. Let's see, if @laurent22 wants the info also on mobile and cli. Sorry for the noise earlier. It's not always that easy to keep track of all the changes.

**Comment by mic704b** [comment]
> Sorry for the noise earlier. It's not always that easy to keep track of all the changes.

All good!

**Comment by laurent22** [comment]
This PR makes me realise there should be a single method to retrieve version number on all platforms. The method could be added to shim.js, then version number would be retrived from package-info on desktop, package.js on CLI and using the Version package on RN.

**Comment by mic704b** [comment]
Ok, I'll put this one on hold for now.

**Pull request closed without merging**